backend/src/utils/resourceUtils.ts

```diff
@@ -133,93 +148,157 @@ const fetchInstalledKfdefs = async (fastify: KubeFastifyInstance): Promise<KfDef
 };
 
 const fetchApplicationDefs = async (fastify: KubeFastifyInstance): Promise<OdhApplication[]> => {
-  const normalizedPath = path.join(__dirname, '../../../data/applications');
   const applicationDefs: OdhApplication[] = [];
   const featureFlags = getComponentFeatureFlags();
-  fs.readdirSync(normalizedPath).forEach((file) => {
-    if (yamlRegExp.test(file)) {
-      try {
-        const doc = jsYaml.load(fs.readFileSync(path.join(normalizedPath, file), 'utf8'));
-        if (!doc.spec.featureFlag || featureFlags[doc.spec.featureFlag]) {
-          applicationDefs.push(doc);
-        }
-      } catch (e) {
-        console.error(`Error loading application definition ${file}: ${e}`);
-      }
-    }
-  });
 
-  const enabledAppsCMData: { [key: string]: string } = {};
-  let changed = false;
-
-  // fetch enabled apps configmap
-  const coreV1Api = fastify.kube.coreV1Api;
-  const namespace = fastify.kube.namespace;
-  const enabledAppsCM: V1ConfigMap = await coreV1Api
-    .readNamespacedConfigMap(enabledAppsConfigMapName, namespace)
-    .then((result) => result.body)
-    .catch(() => null);
+  const customObjectsApi = fastify.kube.customObjectsApi;
+  try {
+    const res = await customObjectsApi.listNamespacedCustomObject(
```

Comment:
There will be an issue if there are more than 250 applications. We should fetch in a loop, similar to fetching subscriptions: https://github.com/opendatahub-io/odh-dashboard/blob/main/backend/src/utils/resourceUtils.ts#L75

Reply:
done.